src/libs/ajax/AzureStorage.ts

```diff
@@ -2,8 +2,23 @@ import _ from 'lodash/fp'
 import { authOpts, fetchOk, fetchWorkspaceManager } from 'src/libs/ajax/ajax-common'
 import { getConfig } from 'src/libs/config'
 import * as Utils from 'src/libs/utils'
-import { getExtension, tools } from 'src/pages/workspaces/workspace/analysis/notebook-utils'
-
+import {
+  AbsolutePath,
+  AnalysisFile,
+  FileMetadata,
+  getDisplayName,
+  getExtension, getFileName
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { cloudProviders } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import { toolLabelTypes, tools } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { cloudProviderTypes } from 'src/pages/workspaces/workspace/workspace-utils'
```

Comment:
hmm... here, we are using some analysis types to decorate API data. It could be in another layer/file and have API-only code and then a separate module to decorate it into analysis-specific types, but this seems a bit overkill as the file is fairly small.

Reply:
this is in fact backwards... and will cause issues as we continue to make code/organizing improvements.  I suggest we flip things.  The types (and helpers) in question should live in the src/libs/ajax subarea, NOT src/pages or src/components.  They are data/BL types, and so should live in the data/BL area, not the visual components areas.